File: src/main/webapp/js/search/search.controller.js
(function () {
    'use strict';

    angular
        .module('searchApp')
        .controller('SearchController', ['searchService', SearchController]);

    function SearchController(searchService) {
        var vm = this;
        vm.searchInput = {search: ''};
        vm.searchResult = [];
        vm.onSearch = onSearch;

        function onSearch() {
            searchService.search(vm.searchInput)
                .then(function (searchResult) {
                    vm.searchResult = searchResult;
                });
        }
    }

})();

File: src/main/java/io/github/perjansson/search/SearchController.java
package io.github.perjansson.search;

import io.github.perjansson.user.User;
import io.github.perjansson.user.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping(value = "/api/search")
public class SearchController {

    private final UserRepository userRepository;

    @Autowired
    public SearchController(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @RequestMapping(method = RequestMethod.POST)
    @ResponseBody
    public List<User> searchUsers(@RequestBody SearchInput searchInput) {
        List<User> searchResult = userRepository.findUsers(searchInput);
        System.out.println(searchInput + " gave " + searchResult);
        return searchResult;
    }

}
